src/layers/linear.rs

```diff
@@ -32,7 +32,7 @@ where
     Tensor<E>: Add<Tensor<Ix1>, Output = Tensor<D>>,
 {
     fn forward(&self, input: &Tensor<D>) -> Tensor<D> {
-        input.dot(&self.weights) + self.biases.clone()
+        input.dot(&self.weights.t().to_owned()) + self.biases.clone()
```

Comment:
Weights needs to be transposed here? Were there issues without `t()`?

Reply:
Yes, without transposing then you will run into panics caused by invalid matrix multiplication. Since we are giving the weight tensor a shape of `[nout, nin]` and the input tensor is of shape `[batches, nin]` then the only way to matmul is with `weights.t()`